core/tests/protractor_utils/AdminPage.js

```diff
@@ -268,6 +306,139 @@ var AdminPage = function() {
       expect(name).toEqual(expectedUsernamesArray[j]);
     }
   };
+
+  this.getMiscTab = async function() {
+    await miscTabButton.click();
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.uploadTopicSimilarities =
+    async function(relativePathToSimilaritiesFile) {
+      absPath = path.resolve(__dirname, relativePathToSimilaritiesFile);
+      await waitFor.visibilityOf(chooseSimilarityFileInput,
+        'Similarity upload form is not visible');
+      await chooseSimilarityFileInput.sendKeys(absPath);
+      await waitFor.elementToBeClickable(similarityFileUploadButton,
+        'Upload button taking too long to be clickable');
+      await similarityFileUploadButton.click();
+    };
+
+  this.expectSimilaritiesToBeUploaded = async function() {
+    await waitFor.textToBePresentInElement(statusMessage,
+      'Topic similarities uploaded successfully.');
+    return true;
+  };
```

Comment:
The waitFor should go in the uploadTopicSimilarities function. This function should be checking that the longer-term effects of the upload happened correctly, e.g. that the UI shows that a file has been uploaded. This expectation should pass even after reloading the page, logging out and back in, etc.

Reply:
I'm not quite sure what you mean here since when I reload the page/log out, the file always disappears from the UI (I'm given the option to upload another file). 

I think that's how it's supposed to be, right?

Reply:
Is there any UI indication that the file has been uploaded? For example, does uploading a file allow the similarities to be downloaded? It's been a while since I've worked with the admin page, so I'm not sure. If you can find a way to confirm that the upload actually worked though, that'd be ideal

Reply:
I've added a line to check that the `similarityDownloadButton` is clickable. I think this will make the test more comprehensive.